### PR TITLE
Remove PLC name annotations

### PR DESCRIPTION
**Pull request by kwxm**
Well that was remarkably tedious.

This removes the annotation type parameter from the `Name` and `TyName` types  to see if that helps with the coercion issue in #1991.  The bad news is that I don't think it does: I'm now getting errors saying "Could not deduce: Coercible uni1 uni".  I'll see if I can fix that.  The serialisation modifications aren't included in this PR: if we merge this then I'll open a new PR for serialisation based on this one.

@reactormonk Could you check that I didn't break the costing code?  I had to make some minor changes to some code involving names (`Cek.hs`).  I think it's OK, but you should make sure.

### CODE REVIEW

File: language-plutus-core/src/Language/PlutusCore/Error.hs [hunk 1]
```diff
@@ -95,15 +95,15 @@ data InternalTypeError uni ann
...
     | TypeMismatch ann
         (Term TyName Name uni ())
         (Type TyName uni ())
         (Normalized (Type TyName uni ()))
     | UnknownDynamicBuiltinName ann UnknownDynamicBuiltinNameError
     | InternalTypeErrorE ann (InternalTypeError uni ann)
-    | FreeTypeVariableE (TyName ann)
-    | FreeVariableE (Name ann)
+    | FreeTypeVariableE TyName
+    | FreeVariableE Name
```
Comment (michaelpj) [inline language-plutus-core/src/Language/PlutusCore/Error.hs:106]: So this does mean we potentially lose information about where the variable was bound in these errors, but I think that's okay. We could add it in if we want it.

Comment (effectfully) [reply language-plutus-core/src/Language/PlutusCore/Error.hs:106]: The other constructors do have annotations, so I think they should be added here as well.

**Comment by kwxm** [comment]
> LGTM

I'm not sure what you mean. Should we have annotations in `FreeTypeVariableE` and `FreeVariableE`?  At the moment these are only used in `TypeCheck/Internal.hs`, where they're called by `lookupTyVar` and `lookupVar` respectively.  These are looking up the names in a `TyVar` and a `Var`, which do contain annotations, so we could rework this a bit to pass the annotations to the errors as well.

**Comment by effectfully** [comment]
> Should we have annotations in `FreeTypeVariableE` and `FreeVariableE`?

That's what I mean, yes. But I don't think it's very important, just makes things nicer and gives better error messages, which nobody is going to read anyway, because they're terrible regardless.

**Comment by kwxm** [comment]
> > Should we have annotations in `FreeTypeVariableE` and `FreeVariableE`?
> 
> That's what I mean, yes. But I don't think it's very important, just makes things nicer and gives better error messages, which nobody is going to read anyway, because they're terrible regardless.

OK.  I'll do that and add an `ann` argument to `lookupTyVar` and `lookupVar`.  Presumably any time we're looking at a name it'll be embedded in something with a sensible annotation, so we should always be able to find something to put in the error.

**Pull request merged**